extensions/functions_arithmetic.yaml

```diff
@@ -362,3 +362,54 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
```

Comment:
I'm not sure this should be solved in this PR immediately because it probably also applies to existing stuff, but something to think about for `SATURATE` in the context of these types of aggregates... What if the internal accumulator overflows but the net result would be within range? For example, `sum:opt_i64(2^63-1, 2^63-1, -2^63-1, -2^63-1)`. Is that *required* to always yield 0? Or for `ERROR`, is it *required* to throw an error (after all, if the second and third element are swapped it wouldn't overflow)?

Reply:
Yes this is a valid point. But I guess this is hard to determine. I would say it is better to leave room for overflows. But I could be wrong.

Reply:
Maybe we should wait with this until we get some more opinions in #251. I'm inclined to say the intermediate value should be the same type, if only because picking i64 feels like an arbitrary "eh, that's probably good enough" decision that doesn't actually solve the problem, but that just makes the overflow question all the more important.

Reply:
Sure, let’s do that.